src/main/java/com/mindee/pdf/PDFUtils.java

```diff
@@ -97,29 +93,80 @@ private static boolean checkIfPdfIsEmpty(PDDocument document) throws IOException
+   * @param source The PDF file.
+   * @return List of all pages as images.
+   */
   public static List<PdfPageImage> pdfToImages(LocalInputSource source) throws IOException {
     PDDocument document = PDDocument.load(source.getFile());
     PDFRenderer pdfRenderer = new PDFRenderer(document);
     List<PdfPageImage> pdfPageImages = new ArrayList<>();
     for (int i = 0; i < document.getNumberOfPages(); i++) {
-      PDRectangle bbox = document.getPage(i).getBBox();
-      float dimension = bbox.getWidth() * bbox.getHeight();
-      int dpi;
-      if (dimension < 200000) {
-        dpi = 300;
-      } else if (dimension < 300000) {
-        dpi = 250;
-      } else {
-        dpi = 200;
-      }
-      BufferedImage imageBuffer = pdfRenderer.renderImageWithDPI(i, dpi, ImageType.RGB);
+      BufferedImage imageBuffer = pdfPageToImageBuffer(i, document, pdfRenderer);
       pdfPageImages.add(new PdfPageImage(imageBuffer, i, source.getFilename(), "jpg"));
     }
     document.close();
     return pdfPageImages;
   }
+
+  /**
+   * Render a single page of a PDF as an image.
+   * Main use case is for processing PDFs with hundreds of pages.
+   * If you need to only render some pages from the PDF, use <code>mergePdfPages</code> and then <code>pdfToImages</code>.
+   * @param filePath The path to the PDF file.
+   * @param pageNumber The page number to render, first page is 1.
+   * @return The page as an image.
+   */
+  public static PdfPageImage pdfPageToImage(String filePath, int pageNumber) throws IOException {
+    return pdfPageToImage(new LocalInputSource(filePath), pageNumber);
+  }
+
+  /**
+   * Render a single page of a PDF as an image.
+   * Main use case is for processing PDFs with hundreds of pages.
+   * If you need to only render some pages from the PDF, use <code>mergePdfPages</code> and then <code>pdfToImages</code>.
+   * @param source The PDF file.
+   * @param pageNumber The page number to render, first page is 1.
+   * @return The page as an image.
+   */
+  public static PdfPageImage pdfPageToImage(
+      LocalInputSource source,
+      int pageNumber
+  ) throws IOException {
+    int index = pageNumber - 1;
+    PDDocument document = PDDocument.load(source.getFile());
+    PDFRenderer pdfRenderer = new PDFRenderer(document);
+    BufferedImage imageBuffer = pdfPageToImageBuffer(index, document, pdfRenderer);
+    return new PdfPageImage(imageBuffer, index, source.getFilename(), "jpg");
+  }
+
+  private static BufferedImage pdfPageToImageBuffer(
+      int index,
+      PDDocument document,
+      PDFRenderer pdfRenderer
+  ) throws IOException {
+    PDRectangle bbox = document.getPage(index).getBBox();
+    float dimension = bbox.getWidth() * bbox.getHeight();
+    int dpi;
+    if (dimension < 200000) {
+      dpi = 300;
+    } else if (dimension < 300000) {
+      dpi = 250;
+    } else {
+      dpi = 200;
+    }
+    return pdfRenderer.renderImageWithDPI(index, dpi, ImageType.RGB);
+  }
```

Comment:
Wouldn't it be more practical to have a default DPI but give a parameter to allow for lower manual values for the user?

Reply:
I believe we had already discussed this for the initial implementation.

The idea here is to make the right decision for the end user. We could add another method allowing the dpi to be set if requested by users.